### PR TITLE
feat: row inserter

### PR DESCRIPTION
**Pull request by fengjiachun**
row based inserter with gRPC

### CODE REVIEW

File: greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ *
+ * @author jiachun.fjc
+ */
+public final class RowHelper {
+
+    public static void addValue(RowData.Row.Builder builder, Common.ColumnDataType dataType, Object value) {
+        RowData.Value.Builder valueBuilder = RowData.Value.newBuilder();
+        switch (dataType) {
+            case INT8:
+                if (value != null) {
```
Comment (MichaelScofield) [inline greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java:33]: Why not move the `if (value != null)` out of the switch block?

**Review by MichaelScofield — COMMENTED** [review]
Can you upgrade the guava version by the way, security vulnerabilities.

File: greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                }
+                break;
+            case STRING:
+                if (value != null) {
+                    valueBuilder.setStringValue((String) value);
+                }
+                break;
+            case DATE:
+                if (value != null) {
+                    valueBuilder.setDateValue((int) value);
```
Comment (killme2008) [inline greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java]: Can not set a `java.util.Date`?

File: greptimedb-protocol/src/main/java/io/greptime/models/TableSchema.java [hunk 3]
```diff
@@ -34,8 +35,8 @@ public class TableSchema {
 
     private TableName tableName;
     private List<String> columnNames;
-    private List<Columns.Column.SemanticType> semanticTypes;
-    private List<Columns.ColumnDataType> dataTypes;
+    private List<Common.SemanticType> semanticTypes;
```
Comment (killme2008) [inline greptimedb-protocol/src/main/java/io/greptime/models/TableSchema.java:38]: We change the packages of `SemanticType` etc. Is the old client still compatible with the DB?

File: greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java (hunk 2 above)
Comment (killme2008) [reply greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java]: Even `LocalDate` etc. Refer to https://www.w3schools.com/java/java_date.asp

File: greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                }
+                break;
+            case DATETIME:
+                if (value != null) {
+                    valueBuilder.setDatetimeValue(Util.getLongValue(value));
```
Comment (killme2008) [inline greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java]: The same question as `Date`.

**Comment by fengjiachun** [comment]
> Can you upgrade the guava version by the way, security vulnerabilities.

Done

**Comment by fengjiachun** [comment]
@MichaelScofield PTAL

**Review by MichaelScofield — APPROVED** [review]

File: greptimedb-protocol/src/main/java/io/greptime/models/TableSchema.java (hunk 3 above)
Comment (fengjiachun) [reply greptimedb-protocol/src/main/java/io/greptime/models/TableSchema.java:38]: I have tested the tags/v0.3.0 version and it is working fine.

File: greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java (hunk 2 above)
Comment (fengjiachun) [reply greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java]: Will be supported in the next PR

File: greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java (hunk 4 above)
Comment (fengjiachun) [reply greptimedb-protocol/src/main/java/io/greptime/models/RowHelper.java]: Will be supported in the next PR

**Pull request merged**